deep_speech_2/KenLMModel.py

```diff
@@ -0,0 +1,112 @@
+import os
+import math
+
+
+class KenLMModel(object):
+    """
+    Wrapper for KenLM language model.
+    You should install KenLM python interface first.
+
+    .. code-block:: python
+    
+        pip install https://github.com/kpu/kenlm/archive/master.zip
+
+    Please refer to **Scalable Modified Kneser-Ney Language Model Estimation** for
+    more details about KenLM
+    """
+
+    def __init__(self,
+                 model_path,
+                 unk_id,
+                 bos_id,
+                 eos_id,
+                 id_str_dict,
+                 verbose=False):
+        """
+        Initialize variables and load model.
+
+        :param model_path: Path of language model
+        :type model_path: str
+        :param unk_id: Identifier for OOV
+        :type unk_id: int
+        :param bos_id: Identifier for start token
+        :type bos_id: int
+        :param eos_id: Identifier for end token
+        :type eos_id: int
+        :param id_str_dict: Dictionary mapping id to word
+        :type id_str_dict: dict
+        :param verbose: Whether print debug information
+        :type verbose: bool
+        """
+        assert unk_id in id_str_dict, 'unk_id must be in id_str_dict'
+        assert bos_id in id_str_dict, 'bos_id must be in id_str_dict'
+        assert eos_id in id_str_dict, 'eos_id must be in id_str_dict'
+        self._model_path = model_path
+        self._unk_id = unk_id
+        self._bos_id = bos_id
+        self._eos_id = eos_id
+        self._id_str_dict = id_str_dict
+        self._verbose = verbose
+
+        self._load_model()
+        if self._verbose:
+            print("Load model done.")
+
+    def _load_model(self):
+        import kenlm
+        self._model = kenlm.LanguageModel(self._model_path)
+
+    def score_sentence_ids(self, id_list):
```

Comment:
A string type is better than an id list. 
And this class should be responsible for tokenization, dictionary loading and dictionary looking-up. Don't leave it to users.